### PR TITLE
Assign picture password when user's last role is deleted

### LINKED ISSUE #14556: Assign picture password when a user's last role is removed
❌ **This issue is not open for contribution. Visit <a href="https://learningequality.org/contributing-to-our-open-code-base/" target="_blank">Contributing guidelines</a>** to learn about the contributing process and how to find suitable issues.

## Overview

When a user's role is removed and they no longer have any remaining roles, they become a learner again. If picture login is enabled for their facility and they do not yet have a `picture_password`, one should be automatically assigned, mirroring the assignment that happens when a new learner is created.

**Complexity:** Low
**Target branch:** develop

### Context

Picture passwords are assigned to learners when picture login is enabled for a facility. As a follow up to #14549, where assigned picture passwords are removed upon role creation, this issue addresses the role-downgrade path: when a coach or admin has their role(s) removed, they revert to being a learner, and need to have a a `picture_password` assigned.

`Role` instances are removed via `Role.delete()` in `kolibri/core/auth/models.py`, which is the method invoked by `RoleResource.deleteCollection({ user: id })` on the frontend as well as any model-layer role removal. 

### The Change

Add a post-delete check inside `Role.delete()`. After the role is deleted, the following conditions must all be true before attempting picture password assignment:

1. The user has no remaining roles (`user.roles.exists()` is `False`)
2. `user.picture_password` is `None`
3. `user.facility.dataset.picture_password_settings` is not `None`
4. The facility learner count is within the 1300-learner capacity limit

If all conditions are met, call `assign_picture_password`. If `NoAvailableSequences` is raised (the sequence pool is exhausted), it should be caught silently, and the `picture_password` should be left unassigned, mirroring the capacity-exceeded behavior at new-learner creation.

### Out of Scope

- Assigning picture passwords to users who still have at least one role
- Any changes to the frontend role-deletion flow
- Handling the case where a learner already has a `picture_password` set
- Bulk backfill for previously downgraded users

#### General
- [ ] After a user's last role is deleted, if `picture_password_settings` is not null and the user has no `picture_password`, one is assigned
- [ ] If the user still has one or more remaining roles after the deletion, no assignment is attempted
- [ ] If `picture_password` is already set, no assignment is attempted
- [ ] If the facility learner count is at or above the 1300-learner limit, no assignment is attempted and no error is raised
- [ ] If `NoAvailableSequences` is raised, it is caught silently and `picture_password` remains `None`
- [ ] Coach and admin users with remaining roles are never assigned a `picture_password`

#### Testing
- [ ] Unit test: role deleted, there are no remaining roles, `picture_password_settings` not null, user under capacity: a `picture_password` is assigned
- [ ] Unit test: role deleted but user still has a remaining role, no picture password is assigned
- [ ] Unit test: role deleted, there are no remaining roles, and `picture_password` already set: no reassignment happens
- [ ] Unit test: role deleted, there are no remaining roles, but the facility at or above 1300 learners: no assignment happens, and no exception is raised
- [ ] Unit test: role deleted, there are no remaining roles: `NoAvailableSequences` is raised by `assign_picture_password`, the exception is caught and no error propagates

### References

- `kolibri/core/auth/models.py` 
- `kolibri/core/auth/utils` `
- `kolibri/core/auth/errors.py` 

## AI usage
🤖 This issue was drafted with AI assistance, under supervision, review, and final edits by [@LianaHarris360]. 🤖

### PR DESCRIPTION
**Pull request by rtibblesbot**
## Summary

After a user's last role is deleted (reverting them to learner status), if picture login is enabled for their facility and they don't yet have a `picture_password`, one is now automatically assigned. This mirrors the assignment that happens at new-learner creation and the clearing that happens at role creation (learningequality/kolibri#14549).

The change adds a post-delete check to `Role.delete()`. `NoAvailableSequences` is caught silently (capacity-exceeded case), following the same pattern as `FacilityUserSerializer.create()`. Inline imports are used inside the method body to avoid a circular import (`picture_passwords.py` imports `Facility`/`FacilityUser` from `models.py`).

## References

Closes learningequality/kolibri#14556

Related: learningequality/kolibri#14549 (clears picture password on role creation — this PR handles the inverse path)

## Reviewer guidance

Five new unit tests in `RoleDeletePicturePasswordTestCase` (`kolibri/core/auth/test/test_api.py`) cover all acceptance criteria: last role deleted assigns password, remaining role prevents assignment, existing password not overwritten, at learner limit no assignment/no exception, `NoAvailableSequences` caught silently.

Area worth checking: the inline imports inside `Role.delete()` — this is an intentional workaround for the circular import described above.

## AI usage

Implemented with Claude Code following a pre-written implementation plan using TDD. Tests were written and confirmed failing before the implementation was added. I reviewed the code, ran the full auth test suite (`pytest kolibri/core/auth/test/`), and verified all tests pass with no regressions.

---
_@rtibblesbot's comments are generated by an LLM, and should be evaluated accordingly_

<details><summary>How was this generated?</summary>

- Ran pre-flight CI checks (lint, format, tests) and verified all pass
- Rebased onto the target branch and resolved any conflicts
- Reorganized commit history into clean, logical commits
- Audited the diff to ensure only issue-relevant files are changed
- Built PR body from the repository's PR template with evidence blocks

</details>

status

### CODE REVIEW

File: packages/kolibri-common/components/__tests__/AllPasswordsPage.spec.js [hunk 1]
```diff
@@ -3,6 +3,8 @@ import FacilityUserResource from 'kolibri-common/apiResources/FacilityUserResour
 import { picturePasswordStrings } from 'kolibri-common/strings/picturePasswords';
 import AllPasswordsPage from '../AllPasswordsPage.vue';
 
+jest.mock('kolibri/composables/useNav');
```
Comment (LianaHarris360) [inline packages/kolibri-common/components/__tests__/AllPasswordsPage.spec.js]: This test file does not need to be updated with this mock for this change, this should be removed.

File: kolibri/core/auth/test/test_api.py [hunk 2]
```diff
@@ -3613,6 +3614,115 @@ def test_prepare_for_bulk_create_uuid_is_deterministic(self):
...
+    def test_no_available_sequences_caught_silently(self, mock_assign):
+        """If assign_picture_password raises NoAvailableSequences, the exception
+        is caught and picture_password remains None."""
+        user, role = self._make_user_with_role("nosequences")
+
+        role.delete()  # must not raise
+
+        user.refresh_from_db()
+        self.assertIsNone(user.picture_password)
+
```
Comment (LianaHarris360) [inline kolibri/core/auth/test/test_api.py:3714]: It would be worth it to include a test case for when a user's last role is deleted in a facility where `picture_password_settings` is None to confirm that they are not assigned a `picture_password`.

**Review by LianaHarris360 — CHANGES_REQUESTED** [review]
The code changes align with the issue specced, but an unrelated change to `AllPasswordsPage.spec.js` should be removed, and a regression test for when a user's last role is deleted in a facility where picture login isn't enable should be added for more thorough testing.

File: packages/kolibri-common/components/__tests__/AllPasswordsPage.spec.js (hunk 1 above)
Comment (rtibblesbot) [reply packages/kolibri-common/components/__tests__/AllPasswordsPage.spec.js]: Done — removed the  line and dropped the entire AllPasswordsPage spec commit. The file is no longer in the diff.

File: kolibri/core/auth/test/test_api.py (hunk 2 above)
Comment (rtibblesbot) [reply kolibri/core/auth/test/test_api.py:3714]: Added — new test `test_last_role_deleted_no_assignment_when_picture_password_settings_null` covers this case. It temporarily disables picture password settings, deletes the user's last role, and asserts `picture_password` remains `None`.

**Review by LianaHarris360 — APPROVED** [review]
